### DISCUSSION TITLE
20221006 회의 내용(코드 베이스 및 게임 구성 회의)

### DISCUSSION THREAD

**Discussion by seonhjeo** [42StudyGame/RandomDiceDefense42-Client#21] [General]
### 목차

1. 전체적인 게임 구성
2. 기능 사용 범위
3. 차후 할 일

### 전체적인 게임 구성
<img width="830" alt="Screen Shot 2022-10-06 at 5 55 35 PM" src="https://user-images.githubusercontent.com/42938826/194268964-02a09854-c313-4b81-b2c0-f6f3347570c7.png">
 - 각각의 객체는 소유와 피소유 관계를 기본으로 연결됩니다.
 - 피소유 관계의 객체는 소유 관계의 객체를 참조하여 소유 객체의 값에 접근할 수 있습니다.

### 기능 사용 범위
 - 각각의 코딩 방식과 언어의 사용 범위 중에서 가장 충돌 및 혼동이 많이 발생할 것 같은 기능들에 대해 앞으로 어떻게 할지 정하였습니다.
   - partial : 성능 및 가독성에 큰 영향을 끼치지 않아 각자 연습 용도로만 적절히 사용하기로 하였습니다.
   - Interface : 각 클래스가 실행할 수 있는 기능을 구분하는 정도로만 사용하기로 하였습니다.
   - scriptable object : 객체의 모든 변수(총알 속도, 주사위의 공격력 등)은 스크립터블 오브젝트로 초기화하기로 하였습니다. 다만 함수 혹은 오브젝트(특수능력, 스프라이트 등)을 스크립터블 오브젝트로 초기화할 수 있는지 여부를 추후 확인해 자식 클래스의 구현 여부를 결정지을 예정입니다.
 
### 차후 할 일
 - 2022년 10월 8일 토요일에 모여 각 객체의 세부적인 기획을 할 예정입니다. 여유가 되면 플레이 외적인 부분(덱메이킹 등)도 토론해볼 예정입니다.

**Comment by seonhjeo** [comment]
틀린 부분이나 추가해야 할 사항이 있으면 댓글에 남겨주세요!
확인 후 수정하겠습니다!

**Reply by giftof** [reply]
scriptableObject 에 추상화된 Spell을 ASpell로 넣어두게되면
public ASpell spell;
처럼 받을수 있어서, spell들만 gameObject로 만들어두고 그걸 넣어주면 되긴 합니다만,
토요일 저녁에, 코드 원형 보여드릴테니, 의견 주시면 될듯요.

**Comment by giftof** [comment]
branch kyuhkim_so < 여기 들어가시면
Module/Tower/Spell/*.*
Module/Tower/Script/TowerObjectData.cs
Module/Tower/Scriptable Object/*.*

이렇게 보시면 됩니다.

실제 실행 했을때 결과는, 브랜치 받아서 실행 시킨 뒤에
Create Deck > Create Tower 실행하면, 부모와 자식이 각각 자기소개하는 Debug를 콘솔창에 띄워두게 했습니다.

**Reply by giftof** [reply]
010-7424-0747

**Reply by giftof** [reply]
얼마나 도움이 될지는 모르겠습니다만, 일단;;; 남겨두어요;;;

**Comment by seonhjeo** [comment]
20221008 추가 회의 내용(수요일까지 할 일)
 - 클래스 정의를 바탕으로 스도코드 작성
 - 작성한 스도코드를 바탕으로 코드 머지
